src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -45,6 +45,7 @@ internal class TagHelperParseTreeRewriter
         private TagHelperBlockTracker _currentTagHelperTracker;
         private BlockBuilder _currentBlock;
         private string _currentParentTagName;
```

Comment:
Instead of this and the `_currentParentIsTagHelper` variable just make use of the tracker stack's current `TagBlockTracker`. We do something simpler with tag helper block trackers (_currentTagHelperTracker).